### DISCUSSION TITLE
SImple question about logging

### DISCUSSION THREAD

**Discussion by RRadziejewski** [pytest-dev/pytest#10568] [Q&A] [answered]
Hello i wan't to get rid of pytest without STDOUT redirection.

So far i'm there : 
![PjRRyF4a5Z](https://user-images.githubusercontent.com/32438378/206424550-a6288f5f-7b1b-4309-9820-0a7dd216a15e.png)

I wan't to get rid off this **green** dot and **green** string. Basically i wan't to get rid of any pytest output without redirection.

**Comment by The-Compiler** [comment]
The string at the end is easy: Just pass another `-q`.

The dot is much harder, as that kind of thing isn't really intended. You can pull it off by [wrapping](https://docs.pytest.org/en/latest/how-to/writing_hook_functions.html#hookwrapper) the default [`pytest_report_teststatus` hook](https://docs.pytest.org/en/latest/reference/reference.html#pytest.hookspec.pytest_report_teststatus) to ignore dots, e.g. like this in a `conftest.py`:

```python
import pytest

@pytest.hookimpl(hookwrapper=True)
def pytest_report_teststatus(report):
    outcome = yield
    category, letter, word = outcome.get_result()
    if letter == ".":
        outcome.force_result((category, "", word))
```

...but with all that said, I can't help but ask: Why?! What's your goal?

**Reply by RRadziejewski** [reply]
Well, you know bash dialog based on n curses? I am using it and simultaneosly running some pytest checks.
If something prints to STDOUT/STDERR (println) in the meantime. My dialog just moves upward which i don't want to happen.

**Comment by nicoddemus** [comment, marked as answer]
`-s -p no:terminal` will do the trick. 👍

**Reply by The-Compiler** [reply]
This would show no output at all even for failing tests and such though - not sure if that's what OP intended.

**Reply by RRadziejewski** [reply]
It was intended, thank you for both the asnwers, and @The-Compiler you are right i did not mention STDERR.
